sys/can/router.c

```diff
@@ -213,13 +212,15 @@ int can_router_register(can_reg_entry_t *entry, canid_t can_id, canid_t mask, vo
     filter_el_t *filter;
     int ret;
 
-#if ENABLE_DEBUG
-    if (entry->type == CAN_TYPE_DEFAULT) {
-        DEBUG("can_router_register: ifnum=%d, pid=%" PRIkernel_pid ", can_id=0x%" PRIx32
-              ", mask=0x%" PRIx32 ", data=%p\n", entry->ifnum, entry->target.pid, can_id, mask, param);
-    } else if (entry->type == CAN_TYPE_MBOX) {
-        DEBUG("can_router_register: ifnum=%d, mbox=%p, can_id=0x%" PRIx32
-              ", mask=0x%" PRIx32 ", data=%p\n", entry->ifnum, (void *)entry->target.mbox, can_id, mask, param);
+#ifdef MODULE_CAN_MBOX
```

Comment:
I don't get why this now should depend on `MODULE_CAN_MBOX`

Reply:
If `MODULE_CAN_MBOX` is defined, then `entry->type` exists.